File: source/frontend/src/lib/dto/request/entity/ArchitectureRationaleRequestDto.ts
export class ArchitectureRationaleRequestDto {
    id : string = "";
    title : string = "";
    context : string = "";
    decision : string = "";
    status : string = "";
    consequences : string = "";

    constructor() {}

    static create() {
        return new ArchitectureRationaleRequestDto();
    }

}

File: source/backend/src/main/java/ch/unisg/backend/controller/http/dto/request/node/ArchitectureRationaleRequestDto.java
package ch.unisg.backend.controller.http.dto.request.node;

import lombok.Getter;

import java.util.UUID;

@Getter
public class ArchitectureRationaleRequestDto {

    public static final String MEDIA_TYPE = "application/architecture-rationale+json";

    private UUID id;
    private String title;
    private String context;
    private String decision;
    private String status;
    private String consequences;
}
